types/parse5/index.d.ts

```diff
@@ -5,7 +5,8 @@
 //                 James Garbutt <https://github.com/43081j>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-export interface Location {
+export {};
```

Comment:
this one still isn't resolved as far as i've seen. we are not declaring things like in what you linked, we export everything, so this line definitely should not be needed.

Reply:
I intentionally don’t want to export the `interface *Base` types, since they’re an implementation detail to simplify **TSDoc** documentation: <https://github.com/DefinitelyTyped/DefinitelyTyped/blob/1136d63651e371d6f7539ed42dc906b3aa9a79bb/types/parse5/parse5-tests.ts#L222-L224>

Reply:
you don't need to export them? just keep them defined internally and export the extensions. you still don't need the `export {}`

Reply:
Removing `export {}` causes the `interface *Base` types to be **_implicitly exported_**, thus causing the above tests to fail with:
```
Expected an error on this line, but found none. (expect)
```

Reply:
wow that is some annoying behaviour, now i finally understand after running dtslint on your branch myself.

i went down the rabbit hole and found andy explaining it.. he does have a point that nobody really has any good reason to hide any types. a consumer could find our base types useful (maybe extending them? etc). so yeah me personally, i'd just expose them, but do as you wish

Reply:
for future reference: andy doesn't work on typescript anymore, and hasn't for a few years now.